src/components/SQForm/useFormButton.tsx

```diff
@@ -3,7 +3,21 @@ import {useFormikContext} from 'formik';
 import {useDebouncedCallback} from 'use-debounce';
 import {hasUpdated} from '../../utils';
 
-export function useFormButton(isDisabled, shouldRequireFieldUpdates, onClick) {
+type UseFormButtonReturnType = {
+  isButtonDisabled: boolean;
+  hasFormBeenUpdated: boolean;
+  values: unknown;
+  initialValues: unknown;
+  isValid: boolean;
+  handleClick: (...args: unknown[]) => void;
```

Comment:
Does this need to be `unknown`?
Could this be the return type of `useDebouncedCallback`?

Reply:
So actually i've been playing around with this after you mentioned it and here are some of the things i've tried,

First i noticed that useDebouncedCallback has a type of `DebouncedState<(event: any) => undefined>` so if i take an approach similar to what Cody suggested below, like this `DebouncedState<(event: React.MouseEventHandler<HTMLButtonElement>)=> void | undefined>;` Then I actually get an error in the SQFormButton component on the getClickHandler function, something like "expecting 1 arg but got 0 or more" which led me to believe that perhaps the getClickHandler function should also not be spread param but instead a single param.

` const getClickHandler = (...args: unknown[]) => {
    if (type === 'reset') {
      return handleReset;
    } else if (typeof onClick !== 'undefined') {
      return handleClick(...args);
    }
  };`

it would look something like this after i adjust it, `const getClickHandler = (args: MouseEventHandler) => {
    if (type === 'reset') {
      return handleReset;
    } else if (typeof onClick !== 'undefined') {
      return handleClick(args);
    }
  };` Though this seems a little narrow, so i was thinking either sticking with the original implimentation of `handleClick: (...args: unknown[]) => void;` or something like DebouncedState<(event: unknown)=> void | undefined>;
  
  what do you guys think?  @SeanGroff @codyarose

Reply:
How about in `UseFormButtonReturnType`:
```ts
handleClick: DebouncedState<React.MouseEventHandler<HTMLButtonElement>>
```
The `useFormButton` `onClick`:
```ts
onClick?: React.MouseEventHandler<HTMLButtonElement>
```
Then pass a generic into `useDebouncedCallback` so it can infer the `event` type:
```ts
const handleClick = useDebouncedCallback<
  React.MouseEventHandler<HTMLButtonElement>
>(event => onClick && onClick(event), 500, {
  leading: true,
  trailing: false
});
```
---
Over in `SQFormButton` the `onClick` prop is wrong, it should be:
```ts
onClick?: React.MouseEventHandler<HTMLButtonElement>
```
Then `getClickHandler` should probably just have an `event` arg:
```ts
const getClickHandler = (event: React.MouseEvent<HTMLButtonElement>) => {
  if (type === 'reset') {
    return handleReset;
  } else if (typeof onClick !== 'undefined') {
    return handleClick(event);
  }
};
```